bftengine/src/bftengine/ReplicaImp.cpp

```diff
@@ -832,15 +837,16 @@ std::pair<PrePrepareMsg *, bool> ReplicaImp::buildPrePrepareMessage() {
     TimeRecorder scoped_timer1(*histograms_.addAllRequestsToPrePrepare);
     ClientRequestMsg *nextRequest = (!requestsQueueOfPrimary.empty() ? requestsQueueOfPrimary.front() : nullptr);
```

Comment:
i see nextRequest is deleted inside addRequestToPrePrepareMessage.
did you check for a new task here + convert requestsQueueOfPrimary to smart?

Reply:
Looks like Tytus converts it as part of his PR (https://github.com/glevkovich/concord-bft/pull/75)